src/backend/src/bitcoin_utils.rs

```diff
@@ -0,0 +1,75 @@
+use std::str::FromStr;
+use ic_cdk::api::management_canister::bitcoin::{BitcoinNetwork};
+use k256::sha2;
+use k256::sha2::Digest;
+
+fn parse_derivation_path(path: &str) -> Vec<Vec<u8>> {
+    let re = regex::Regex::new(r"^m/44'(/\d+'?){2}/[01]/\d+$").unwrap();
+    assert!(re.is_match(path), "Invalid BIP-44 derivation path format");
+
+    path.split('/')
+        .skip(1) // Skip the leading 'm'
+        .map(|s| {
+            if s.ends_with('\'') {
+                // Handle hardened derivation (apostrophe indicates hardened)
+                // Value 0x80000000 is added to represent a hardened key.
+                // It is the hexadecimal of the smallest 32-bit value with the high-order bit set
+                // More details: https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#Extended_keys
+                let index = u32::from_str(&s[..s.len() - 1]).unwrap() + 0x80000000;
+                index.to_be_bytes().to_vec()
+            } else {
+                let index = u32::from_str(s).unwrap();
```

Comment:
How does `u32::from_str()` handle leading zeros?  Does it accept them or return an error?

Reply:
last changes deprecated the usage of `u32::from_str()`